pkg/filters/string.go

```diff
@@ -245,12 +244,28 @@ func (filter *BPFStringFilter) InitBPF(bpfModule *bpf.Module) error {
 		return err
 	}
 
+	filterVal := make([]byte, 16)
+
 	//Initialize the associated bpfMap
 	//First initialize notEqual values since equality should take precedence
 	for str := range filter.notEqual {
 		byteStr := make([]byte, maxBpfStrFilterSize)
 		copy(byteStr, str)
-		if err := bpfMap.Update(unsafe.Pointer(&byteStr[0]), unsafe.Pointer(&bpfFilterNotEqual)); err != nil {
+
+		var equalInScopes, equalitySetInScopes uint64
+		curVal, err := bpfMap.GetValue(unsafe.Pointer(&byteStr[0]))
+		if err == nil {
```

Comment:
What if `err != nil`? There's no check for that.

Reply:
Failure is the right result on the first key lookup as we haven't stored it yet. So we could check curVal == nil instead, but it would rule out the other check, logically speaking, because if err != nil, then curVal == nil, and vice versa.

That's why I opted to check err == nil only; we'll always get an err at the first lookup of some key, and it's alright.

But it will be replaced soon with the pre-computed approach as we discussed.